src/java/org/apache/cassandra/index/sai/plan/StorageAttachedIndexSearcher.java

```diff
@@ -190,6 +162,35 @@ private FilterTree analyzeFilter()
         return controller.buildFilter();
     }
 
+    @VisibleForTesting
+    public final Set<String> plannedIndexes()
+    {
+        try
+        {
+            Plan plan = controller.buildPlan();
+            Set<String> indexes = new HashSet<>();
+            plan.forEach(node -> {
+                if (node instanceof Plan.IndexScan)
+                {
+                    Plan.IndexScan indexScan = (Plan.IndexScan) node;
+                    indexes.add(indexScan.getIndexName());
+                }
+                if (node instanceof Plan.ScoredIndexScan)
+                {
+                    Plan.ScoredIndexScan indexScan = (Plan.ScoredIndexScan) node;
+                    indexes.add(indexScan.getIndexName());
```

Comment:
bug fix?

Reply:
Before we were doing that only for IndexScan node, now we have also ScoredIndexScan if I did not misread something.